src/compiler/types.ts

```diff
@@ -400,6 +411,16 @@ module ts {
         HasAggregatedChildData = 1 << 8
     }
 
+    export const enum JsxFlags {
+        None = 0,
+        IntrinsicNamedElement = 1 << 0,
+        IntrinsicIndexedElement = 1 << 1,

```

Comment:
An element that comes from the string indexer of `IntrinsicElements` rather than a named element. We need to distinguish them later on.